pallene/driver.lua

```diff
@@ -83,26 +78,29 @@ function driver.compile_internal(filename, stop_after)
 end
 
 local function compile_pallene_to_c(pallene_filename, c_filename, mod_name)
-    local ok, err, errs
+    local input, err = driver.load_input(pallene_filename)
+    if not input then
+        return false, { err }
+    end
 
-    local module
-    module, errs = driver.compile_internal(pallene_filename)
+    local module, errs = driver.compile_internal(pallene_filename, input)
     if not module then
         return false, errs
     end
 
     local c_code
     c_code, errs = coder.generate(module, mod_name)
     if not c_code then
-        return c_code, errs
+        return false, errs
     end
 
+    local ok
     ok, err = util.set_file_contents(c_filename, c_code)
     if not ok then
-        return ok, {err}
+        errs = { err }
```

Comment:
This should be `return false, {err}`

Reply:
Okay.

But then if `not ok` evaluates to true, doesn't that mean  `ok` is `false`, or at least a falsy value?